File: 523-continuous-subarray-sum/continuous-subarray-sum.py
class Solution(object):
    def checkSubarraySum(self, nums, k):
     d={0:-1}
     sum=0
     for i in range(len(nums)):
        sum+=nums[i];
        if sum%k in d:
            if((i-d[sum%k])>=2):
               return True
        else:
            d[sum%k]=i
     return False
        

File: 523-continuous-subarray-sum/continuous-subarray-sum.java
class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
       HashMap<Integer,Integer> freq=new HashMap<>();
       freq.put(0,-1);
       int sum=0;
       int i=0;
       for(int x:nums){
        sum+=x;
        if(freq.containsKey(sum%k)){
            if(i-freq.get(sum%k)>=2) return true;
        }
        else
          freq.put(sum%k,i);
        i++;
       }
    return false;
    }
}